### ISSUE TITLE
No RSS generated

### ISSUE THREAD

**Issue by hiperterminal** [jgonyea/grav-plugin-podcast#35]
Hi!

I am using this plugin [for my website](http://hiperterminal.com/septetocamajan/) but there is no RSS. I am using podcast v2.1.7 on Grav v1.6.31 

Any idea of what it can be? 

Thanks in advance!!

**Comment by jgonyea** [comment]
Hi! There should be two other dependencies for the podcast plugin.  Have you verified that GetID3 and the Feed plugin are installed?

**Comment by hiperterminal** [comment]
> Have you verified that GetID3 and the Feed plugin are installed?

Yes,  GetId3 v 1.0.3 and Feed v 1.8.0. Both enabled.

**Comment by jgonyea** [comment]
Are you putting your episodes underneath a podcast channel page?

**Comment by hiperterminal** [comment]
> Are you putting your episodes underneath a podcast channel page?

Yes, I am following the sequence Chanel > Series > Episodes.

**Comment by jgonyea** [comment]
Which page are you on where you're trying to find the rss feed?

**Comment by jgonyea** [comment]
Oh, I see, it's something most likely due to the language selection.  Let me look at how I can do that.

If I manually make the URL contain the language (e.g. 'es'), then the rss feed file is available.

**Comment by jgonyea** [comment]
I will be pushing a fix in Podcast 2.1.9 for multilanguage by the end of the day.




I think you might need to add the lines from this section:
https://learn.getgrav.org/17/content/multi-language#multi-language-basics

Specifically, the one about including the default language.

**Comment by jgonyea** [comment]
I just published 2.1.9.  Try updating to that and see if that fixes your issue.

**Comment by hiperterminal** [comment]
> Try updating to that and see if that fixes your issue.

Thanks!! it works for the series, but not for the channel.

**Comment by jgonyea** [comment]
I found the issue.  In the feed.rss.twig, I was looking for file name specifically `podcast-channel.md`, which wouldn't find multi-language pages (podcast-channel.es.md).  Changed the check to look for page.template, and the feed should now be functioning properly.

Fixed in v2.1.10.  Please let me know if it works for you.

**Comment by hiperterminal** [comment]
> Fixed in v2.1.10. Please let me know if it works for you.

Here what happened:

* **Channel**:
  * The rss link at the channel page appears, now it says "Podcast channel | <page name>" which is good, but the link is broken, it aims to /es/.rss (and should be /es/<page name>.rss)
  * If I try to read the channel rss (/es/<page name>.rss) in a feed reader it does not work
* **Series**:
  * The rss link at the series page appears.
  * The rss link is working in a feed reader. 
  * Everything is OK now with the series.

Many thanks!

**Comment by jgonyea** [comment]
I had to clear my local cache, but when I visit your channel page (http://hiperterminal.com/septetocamajan/es), I'm able to click on the rss feed and it shows me an RSS feed.

```
<?xml version="1.0" encoding="UTF-8"?>
<rss version="2.0" xmlns:itunes="http://www.itunes.com/dtds/podcast-1.0.dtd">
<channel>
<title>Podcast</title>
<link>https://www.hiperterminal.com/septetocamajan</link>
<language>es</language>
<copyright>2012 - Septeto Camaján</copyright>
<itunes:subtitle>Conversaciones sobre música</itunes:subtitle>
<itunes:author>El Septeto Camaján</itunes:author>
<itunes:summary>Nuestras emisiones:</itunes:summary>
<description>Transmitiendo desde Bogotá, Colombia.</description>
<itunes:owner>
    <itunes:name>El Septeto Camaján</itunes:name>
    <itunes:email>david@hiperterminal.com</itunes:email>
</itunes:owner>
<itunes:image href="http://hiperterminal.com/septetocamajan/es/"/>
<itunes:category text="Society &amp; Culture">
    <itunes:category text="Music"/>
</itunes:category>
<itunes:explicit>no</itunes:explicit>
<item>
    <title>Grunge</title>
    <link>http://hiperterminal.com/septetocamajan/es/podcast/el-septeto-camajan/grunge</link>
        <itunes:episode>1</itunes:episode>
        <itunes:author>El Septeto Camaján</itunes:author>
    <itunes:subtitle></itunes:subtitle>
    <itunes:summary>Juan Gabriel nos lleva a los 90's en un recorrido muy personal.

Fui muy específico. Debí haber dicho rock de los 90's, pero&#x2026;</itunes:summary>
    <itunes:image href="http://hiperterminal.com/septetocamajan/es/"/>
        <enclosure length="67505422" type="audio/mpeg" url="http://hiperterminal.com/septetocamajan/user/pages/01.podcast/01.el-septeto-camajan/grunge/01-2021-01-12-septeto-camaja-grunge.mp3"/>
    <guid>http://hiperterminal.com/septetocamajan/user/pages/01.podcast/01.el-septeto-camajan/grunge/01-2021-01-12-septeto-camaja-grunge.mp3</guid>
        <pubDate>Tue, 12 Jan 2021 08:21:00 -0500</pubDate>
    <itunes:duration>1:10:19</itunes:duration>
    <itunes:explicit>no</itunes:explicit>
</item>
<item>
    <title>Bandas de 3 y de 4</title>
    <link>http://hiperterminal.com/septetocamajan/es/podcast/el-perro-el-cerdo-y-la-oveja/bandas-de-3-y-de-4</link>
        <itunes:episode>2</itunes:episode>
        <itunes:author></itunes:author>
    <itunes:subtitle></itunes:subtitle>
    <itunes:summary>Conversaciones sobre tríos y cuartetos de rock.
Las bandas

Soda Stereo – Té para tres
Los Tres – Traje de sastre
Ángel Parra&#x2026;</itunes:summary>
    <itunes:image href="http://hiperterminal.com/septetocamajan/es/user/pages/01.podcast/02.el-perro-el-cerdo-y-la-oveja/bandas-de-3-y-de-4/01-logo-PerroCerdoOveja.jpg"/>
        <enclosure length="58796973" type="audio/mpeg" url="http://blog.hiperterminal.com/wp-content/uploads/2012/10/02-PerroCerdoOveja-2012-09-21.mp3"/>
    <guid>http://blog.hiperterminal.com/wp-content/uploads/2012/10/02-PerroCerdoOveja-2012-09-21.mp3</guid>
        <pubDate>Tue, 12 Jan 2021 07:11:00 -0500</pubDate>
    <itunes:duration>1:01:15</itunes:duration>
    <itunes:explicit>no</itunes:explicit>
</item>
<item>
    <title>Los conciertos perdidos</title>
    <link>http://hiperterminal.com/septetocamajan/es/podcast/el-perro-el-cerdo-y-la-oveja/los-conciertos-perdidos</link>
        <itunes:episode>1</itunes:episode>
        <itunes:author></itunes:author>
    <itunes:subtitle></itunes:subtitle>
    <itunes:summary>¿Qué concierto le dolió haberse perdido? es la pregunta que da forma al primer episodio de El perro, el cerdo y la oveja.&#x2026;</itunes:summary>
    <itunes:image href="http://hiperterminal.com/septetocamajan/es/user/pages/01.podcast/02.el-perro-el-cerdo-y-la-oveja/bandas-de-3-y-de-4/01-logo-PerroCerdoOveja.jpg"/>
        <enclosure length="67575204" type="audio/mpeg" url="http://blog.hiperterminal.com/wp-content/uploads/2012/09/01-PerroCerdoOveja-2012-09-11.mp3"/>
    <guid>http://blog.hiperterminal.com/wp-content/uploads/2012/09/01-PerroCerdoOveja-2012-09-11.mp3</guid>
        <pubDate>Tue, 12 Jan 2021 06:04:00 -0500</pubDate>
    <itunes:duration>56:19</itunes:duration>
    <itunes:explicit>no</itunes:explicit>
</item>
</channel>
</rss>

```

**Comment by jgonyea** [comment]
Also, thanks for your patience on this.  I'm not a multi-lingual speaker, so I appreciate your help in diagnosing issues with this.

**Comment by hiperterminal** [comment]
I was trying the feed [in this web](https://rssatom.com/) and I imagine they can not read it because there is no name in the file, just the format (.rss). I open the link in http://hiperterminal.com/septetocamajan/es and yes, I can see the same in a notepad.

And thank you very much for that note about patience. I am a little bit ashamed because I can not change the code yet (it is my first Grav website) but I am very happy to help somehow. I sent other tickets to another plugin and no response yet, I am very happy with your fast answers, thanks!!

**Comment by jgonyea** [comment]
Can you send me your languages section from your system.yaml?  I just rebuilt a site and I'm seeing urls like this:
hostname/es/septetocamjan 

vs yours which has the language at the end
hostname/septetocmjan/es

**Comment by hiperterminal** [comment]
I am not sure if you need this, let me know if is another thing:

```
title: 'El septeto Camaján'
default_lang: es
author:
  name: 'David Ramírez-Ordóñez'
  email: david@hiperterminal.com
taxonomies:
  - category
  - tag
metadata:
  description: 'Podcast sobre música. Analizamos los temas y bandas para tener una perspectiva muy personal de la música que nos gusta.'
summary:
  enabled: true
  format: short
  size: 300
  delimiter: '***'
redirects: null
routes: null
blog:
  route: /blog

```

**Comment by jgonyea** [comment]
No, I was looking for your language settings from the system.yaml file.

**Comment by jgonyea** [comment]
I'm doing a major overhaul to hopefully better support multilang in Grav 1.7.  I expect it to be finished this weekend.

**Comment by hiperterminal** [comment]
> No, I was looking for your language settings from the system.yaml file.

Sorry, I think is this one, if no, let me know:

```
absolute_urls: false
timezone: America/Bogota
default_locale: null
param_sep: ':'
wrapped_site: false
reverse_proxy_setup: false
force_ssl: false
force_lowercase_urls: true
custom_base_url: null
username_regex: '^[a-z0-9_-]{3,16}$'
pwd_regex: '(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,}'
intl_enabled: true
http_x_forwarded:
  protocol: true
  host: false
  port: true
  ip: true
languages:
  supported:
    - es
  default_lang: null
  include_default_lang: true
  pages_fallback_only: false
  translations: true
  translations_fallback: true
  session_store_active: false
  http_accept_language: false
  override_locale: false
home:
  alias: /podcast
  hide_in_urls: false
pages:
  theme: quark
  order:
    by: default
    dir: asc
  list:
    count: 20
  dateformat:
    default: 'Y-m-d H:i'
    short: 'l jS \of F g:i A'
    long: 'F jS \a\t g:ia'
  publish_dates: true
  process:
    markdown: true
    twig: false
  twig_first: false
  never_cache_twig: false
  events:
    page: true
    twig: true
  markdown:
    extra: false
    auto_line_breaks: false
    auto_url_links: false
    escape_markup: false
    special_chars:
      '>': gt
      '<': lt
    valid_link_attributes:
      - rel
      - target
      - id
      - class
      - classes
  types:
    - html
    - htm
    - xml
    - txt
    - json
    - rss
    - atom
  append_url_extension: null
  expires: 604800
  cache_control: null
  last_modified: false
  etag: false
  vary_accept_encoding: false
  redirect_default_route: false
  redirect_default_code: '302'
  redirect_trailing_slash: true
  ignore_files:
    - .DS_Store
  ignore_folders:
    - .git
    - .idea
  ignore_hidden: true
  hide_empty_folders: false
  url_taxonomy_filters: true
  frontmatter:
    process_twig: false
    ignore_fields:
      - form
      - forms
cache:
  enabled: true
  check:
    method: file
  driver: auto
  prefix: g
  purge_at: '0 4 * * *'
  clear_at: '0 3 * * *'
  clear_job_type: standard
  clear_images_by_default: true
  cli_compatibility: false
  lifetime: 604800
  gzip: false
  allow_webserver_gzip: false
  redis:
    socket: null
    server: null
    port: null
    password: null
  memcache:
    server: null
    port: null
  memcached:
    server: null
    port: null
twig:
  cache: true
  debug: true
  auto_reload: true
  autoescape: false
  undefined_functions: true
  undefined_filters: true
  umask_fix: false
assets:
  css_pipeline: false
  css_pipeline_include_externals: true
  css_pipeline_before_excludes: true
  css_minify: true
  css_minify_windows: false
  css_rewrite: true
  js_pipeline: false
  js_pipeline_include_externals: true
  js_pipeline_before_excludes: true
  js_minify: true
  enable_asset_timestamp: false
  collections:
    jquery: 'system://assets/jquery/jquery-2.x.min.js'
errors:
  display: 1
  log: true
log:
  handler: file
  syslog:
    facility: local6
debugger:
  enabled: false
  shutdown:
    close_connection: true
  twig: true
images:
  default_image_quality: 85
  cache_all: false
  cache_perms: '0755'
  debug: false
  auto_fix_orientation: false
  seofriendly: false
media:
  enable_media_timestamp: false
  unsupported_inline_types: null
  allowed_fallback_types: null
  auto_metadata_exif: false
  upload_limit: 33554432
session:
  enabled: true
  initialize: true
  timeout: 1800
  name: grav-site
  uniqueness: path
  secure: false
  httponly: true
  split: true
  path: null
gpm:
  releases: stable
  proxy_url: null
  method: auto
  verify_peer: true
  official_gpm_only: true
accounts:
  type: data
  storage: file
strict_mode:
  yaml_compat: true
  twig_compat: true
  blueprint_compat: true
```

**Comment by jgonyea** [comment]
Perfect, thanks.

**Comment by jgonyea** [comment]
Version 3.0.0 was just released.  Can you take a look and see if the issues are now resolved?

**Comment by hiperterminal** [comment]
There is a Twig error and the whole website is down. [Here the website](http://hiperterminal.com/septetocamajan/es) showing the error.

Thanks!

**Comment by jgonyea** [comment]
Whatever page that's on, can you reopen it and resave it?

**Comment by jgonyea** [comment]
I think I see one issue. Line 13 shouldn't have a '|first'. I'll push 3.0.1 in a handful of hours.

**Comment by jgonyea** [comment]
@hiperterminal , what version of Grav core and admin are you on?  You should be on the latest of both

**Comment by hiperterminal** [comment]
> what version of Grav core and admin are you on?

It's the latest of both: Grav v1.7.3 - Admin v1.10.2

**Comment by jgonyea** [comment]
Could you send me your user/pages folder zipped up, minus any audio?  I can't replicate the issue anymore.

**Comment by hiperterminal** [comment]
Sure! here you have the [pages.zip](https://github.com/jgonyea/grav-plugin-podcast/files/5880117/pages.zip)

**Comment by jgonyea** [comment]
Found the issue!

It has to do with the breaking changes of the image field I had to make to get around the Grav 1.6 file field -> Grav 1.7 filepicker field issue.

To manually fix, open each page that has an image.  If you're doing this locally, it probably will be faster to open the *.md files in a text editor.  

You'll see something along the lines like:
```
series:
    image:
        user/pages/01.podcast/01.el-septeto-camajan/sonido.jpg:
            name: sonido.jpg
            type: image/jpeg
            size: 8206
            path: user/pages/01.podcast/01.el-septeto-camajan/sonido.jpg
```

This needs to be changed to:
```
series:
    image: sonido.jpg
```
Save the file in your editor and re-browse (don't just refresh) to that page in the admin.  It should load properly.

I'll look into how I might be able to automatically manipulate pages as it loads the frontmatter, but before it tries to populate fields.
